SU2_CFD/src/solver_direct_mean_inc.cpp

```diff
@@ -4331,16 +4331,27 @@ void CIncEulerSolver::SetFarfield_AoA(CGeometry *geometry, CSolver **solver_cont
 
 void CIncEulerSolver::Evaluate_ObjFunc(CConfig *config) {
 
-  unsigned short iMarker_Monitoring;
+  unsigned short iMarker_Monitoring, Kind_ObjFunc;
   su2double Weight_ObjFunc;
 
+  Total_ComboObj = 0.0;
+  
   /*--- Loop over all monitored markers, add to the 'combo' objective ---*/
 
   for (iMarker_Monitoring = 0; iMarker_Monitoring < config->GetnMarker_Monitoring(); iMarker_Monitoring++) {
 
```

Comment:
I have been working on updating related code (https://github.com/su2code/SU2/pull/494) - other mismatches between markers and objectives in the combined objective case are taken care of in config_structure.cpp ~ line 2268. I think that a fix for this case can be added there instead, both for consistency and to avoid repeated code (especially since this problem effects the continuous adjoint as well).
I don't mind adding that in to my current pull request; however just a note that that will make checking for the number of markers here redundant. (They won't conflict with each other, just be doing the same thing twice)

Reply:
We need a regression test with several surfaces to be sure that everything is working fine. You can remove this line in your pull request after checking that multiple surfaces are working just fine.

Reply:
I agree, could you please add one that will fail if the functionality addressed by this section of code is removed? Based on other discussion on this pull request, you may be able to hit several birds with one stone by also making that an optimization case using a turbulent flow and the discrete adjoint, and which uses both the OPT_GRADIENT_FACTOR and OPT_RELAX_FACTOR. Trust me, it's better to go through the time to add a regression test early rather than try to figure out what happened to your features months or years down the line.

Reply:
Done!